src/main/java/com/gluonhq/substrate/target/LinuxLinkerFlags.java

```diff
@@ -0,0 +1,190 @@
+public class LinuxLinkerFlags {
+    private static final Flavor flavor = new LinuxFlavor().getFlavor();
+
+    /**
+     * Defines per-flavor pkg-config package names and the associated OS package
+     * name that provides it.
+     */
+    private static final List<PkgInfo> LINK_DEPENDENCIES = List.of(
+        hardwired("-Wl,--no-whole-archive"),
+
+        activeOf(debian("gl", "libgl-dev"),
+                 fedora("gl", "mesa-libGL-devel")),
+        activeOf(debian("x11", "libx11-dev"),
+                 fedora("x11", "libX11-devel")),
+        activeOf(debian("gtk+-x11-3.0", "libgtk-3-dev"),
+                 fedora("gtk+-3.0", "gtk3-devel")),
+        activeOf(debian("freetype2", "libfreetype6-dev"),
+                 fedora("freetype2", "freetype-devel")),
+        activeOf(debian("pangoft2", "libpango1.0-dev"),
+                 fedora("pangoft2", "pango-devel")),
+
+        hardwired("-lgstreamer-lite"),
+
+        activeOf(debian("gthread-2.0", "libglib2.0-dev"),
+                 fedora("gthread-2.0", "glib2-devel")),
+
+        hardwired("-lstdc++"),
+        
+        activeOf(debian("zlib", "zlib1g-dev"),
+                 fedora("zlib", "zlib-devel")),
+
+        activeOf(debian("xtst", "libxtst-dev"),
+                 fedora("xtst", "libXtst-devel")),
+
+        // On fedora these require https://rpmfusion.org/
+        activeOf(debian("libavcodec", "libavcodec-dev"),
+                 fedora("libavcodec", "ffmpeg-devel")),
+        activeOf(debian("libavformat", "libavformat-dev"),
+                 fedora("libavformat", "ffmpeg-devel")),
+        activeOf(debian("libavutil", "libavutil-dev"),
+                 fedora("libavutil", "ffmpeg-devel")),
+
+        activeOf(debian("alsa", "libasound2-dev"),
+                 fedora("alsa", "alsa-lib-devel")),
+
+        hardwired("-lm"),
+
+        activeOf(debian("gmodule-no-export-2.0", "libglib2.0-dev"),
+                 fedora("gmodule-no-export-2.0", "glib2-devel"))
+    );
+
+    public List<String> getLinkerFlags() {
+        List<String> missingPackages = new ArrayList<>();
+
+        List<String> pkgFlags = LINK_DEPENDENCIES.stream()
+                .flatMap(pkg -> lookupPackageFlags(pkg, missingPackages).stream())
+                .collect(Collectors.toList());
+
+        if (!missingPackages.isEmpty()) {
+            printUpdateInstructionsAndFail(missingPackages);
+        }
+
+        Logger.logDebug("All flags: " + pkgFlags);
+        return pkgFlags;
+    }
+
+    /**
+     * Uses pkg-config to lookup linker flags for package.
+     * 
+     * If pkg-config fails, adds amendment instructions to missingPackages.
+     */
+    private List<String> lookupPackageFlags(PkgInfo pkgInfo, List<String> missingPackages) {
+        if (pkgInfo.hardwired != null) {
+            return List.of(pkgInfo.hardwired);
+        }
+
+        String pkgName = pkgInfo.pkgName;
+        ProcessRunner process = new ProcessRunner("/usr/bin/pkg-config", "--libs", pkgName);
+        try {
```

Comment:
Usually we run the process without try-catch and send the possible exceptions to the caller, mostly if these are somehow critical and the process should stop.

Reply:
Not possible because I call it from a stream.
I can use UncheckedIOException instead, if that makes a difference.
Or rewrite to avoid stream, if you prefer that.

Reply:
Yes, the latter. We have done the same (sadly), moving some streams to for-loops.